### PR TITLE
[DOXIA-709] Cleanup overloaded Doxia 1.0 methods

### PR DESCRIPTION
**Pull request by kwin**
Centrally delegate from Doxia 1.0 to Doxia 1.1 or 2.0 equivalents in AbstractSink.
Clarify in javadoc of Sink.
Only implement Doxia 1.1/2.0 variants in Sinks.

### CODE REVIEW

**Comment by kwin** [comment]
This allows to get rid of 1750 lines of code :-)

**Comment by michael-o** [comment]
> This allows to get rid of 1750 lines of code :-)

Man, this I love!

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractSink.java [hunk 1]
```diff
@@ -23,12 +23,337 @@
 
 /**
  * An abstract base class that defines some convenience methods for sinks.
+ * Also acts as compatibility bridge for Doxia 1.0 methods which have overridden variants in Doxia > 1.0
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractSink.java:26]: So this is temporary?

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventTestingSink.java [hunk 2]
```diff
@@ -803,11 +488,6 @@ public void division(SinkEventAttributes attributes) {
...
     }
 
-    @Override
-    public void verbatim() {
-        addEvent("verbatim");
-    }
-
     @Override
     public void verbatim(SinkEventAttributes attributes) {
         addEvent("verbatim", new Object[] {attributes});
```
Comment (michael-o) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventTestingSink.java:493]: These changes I do not fully understand. Now the events are gone?

**Review by michael-o — COMMENTED** [review]
Did you run this down the chain until Maven Site Plugin? Any surprises?

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractSink.java (hunk 1 above)
Comment (kwin) [reply doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractSink.java:26]: As long as the methods not taking SinkEventAttributes are not deprecated, this is permanent.

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventTestingSink.java (hunk 2 above)
Comment (kwin) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventTestingSink.java:493]: No, the events are only added through the methods with the SinkEventAttributes arguments. The other ones just internally delegate to those (via the AbstractSink).

**Comment by kwin** [comment]
> Did you run this down the chain until Maven Site Plugin? Any surprises?

No I haven't yet.

**Review by michael-o — APPROVED** [review]
No objections any more, I trust you here doing the right thing.

**Pull request merged**